src/wallet/walletdb.cpp

```diff
@@ -371,7 +372,8 @@ ReadKeyValue(CWallet* pwallet, CDataStream& ssKey, CDataStream& ssValue,
             CWalletTx wtx;
             ssValue >> wtx;
             CValidationState state;
-            if (!(CheckTransaction(wtx, state) && (wtx.GetHash() == hash) && state.IsValid()))
+            const uint64_t nMaxTxSize = consensusParams.MaxTxSize(wtx.GetTxTime());
```

Comment:
Not sure about that.
The `wtx.GetTxTime()` can response with the stored smartTime and maybe there are cases where not-yet-broadcasted transactions might end up in a `return false;` and therefore are not added tho the in-memory wtx map.

I'm also not so sure if the `CheckTransaction()` call makes sense when reading already stored wtx from the disk. A simple check could help to make sure we don't load wtx from a different net (testnet3 as example),...